# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: media/webrtc/signaling/src/media-conduit/WebrtcMediaDataDecoderCodec.h
# Commit: ffdbeba93fa6
# Full Hash: ffdbeba93fa632a830138cb79c53cfc15bf1f13d
# Author: Dan Minor <dminor@mozilla.com>
# Date: 2018-11-03 09:44:53
# Regressor Bug: 1376873
# File Overlap Count: 1
# Description:
#   Bug 1376873 - WebrtcMediaDataDecoderCodec updates; r=pehrsons
#   
#   This makes the ImageBuffer class implement VideoFrameBuffer class and makes
#   it available in MediaPipeline. To avoid exposing MediaPipeline to a lot of
#   internal details of WebrtcMediaDataDecoder, ImageBuffer is moved to its
# ==============================================================================

diff -r d12ad321047d -r ffdbeba93fa6 media/webrtc/signaling/src/media-conduit/WebrtcMediaDataDecoderCodec.h
--- a/media/webrtc/signaling/src/media-conduit/WebrtcMediaDataDecoderCodec.h	Tue Aug 21 13:40:10 2018 -0400
+++ b/media/webrtc/signaling/src/media-conduit/WebrtcMediaDataDecoderCodec.h	Tue Aug 21 13:40:45 2018 -0400
@@ -9,6 +9,7 @@
 #include "MediaInfo.h"
 #include "MediaResult.h"
 #include "PlatformDecoderModule.h"
+#include "WebrtcImageBuffer.h"
 #include "webrtc/common_video/include/video_frame_buffer.h"
 #include "webrtc/modules/video_coding/include/video_codec_interface.h"
 
@@ -25,16 +26,6 @@
 class SharedThreadPool;
 class TaskQueue;
 
-class ImageBuffer : public webrtc::NativeHandleBuffer
-{
-public:
-  explicit ImageBuffer(RefPtr<layers::Image>&& aImage);
-  rtc::scoped_refptr<VideoFrameBuffer> NativeToI420Buffer() override;
-
-private:
-  RefPtr<layers::Image> mImage;
-};
-
 class WebrtcMediaDataDecoder : public WebrtcVideoDecoder
 {
 public: